src/Selenium2Driver.php

```diff
@@ -297,46 +246,19 @@ private function executeJsOnElement(Element $element, $script, $sync = true)
     public function start()
     {
         try {
-            $this->wdSession = $this->webDriver->session($this->browserName, $this->desiredCapabilities);
-            $this->applyTimeouts();
+            $this->webDriver = RemoteWebDriver::create($this->wdHost, $this->desiredCapabilities);
+            if (\count($this->timeouts)) {
+                $this->applyTimeouts();
+            }
         } catch (\Exception $e) {
-            throw new DriverException('Could not open connection: '.$e->getMessage(), 0, $e);
+            throw new DriverException('Could not open connection: ' . $e->getMessage(), 0, $e);
         }
 
-        if (!$this->wdSession) {
+        if (!$this->webDriver) {
             throw new DriverException('Could not connect to a Selenium 2 / WebDriver server');
         }
-        $this->started = true;
-    }
-
-    /**
-     * Sets the timeouts to apply to the webdriver session
-     *
-     * @param array $timeouts The session timeout settings: Array of {script, implicit, page} => time in milliseconds
-     *
-     * @throws DriverException
-     */
-    public function setTimeouts($timeouts)
-    {
-        $this->timeouts = $timeouts;
-
-        if ($this->isStarted()) {
-            $this->applyTimeouts();
-        }
-    }
 
-    /**
-     * Applies timeouts to the current session
-     */
-    private function applyTimeouts()
-    {
-        try {
-            foreach ($this->timeouts as $type => $param) {
-                $this->wdSession->timeouts($type, $param);
-            }
-        } catch (UnknownError $e) {
-            throw new DriverException('Error setting timeout: ' . $e->getMessage(), 0, $e);
-        }
+        $this->started = true;
```

Comment:
I suggest dropping that boolean entirely, relying on `null !== $this->webDriver` instead to check for being started

Reply:
agree